src/formulas/parser.ts

```diff
@@ -12,11 +12,42 @@ const UNARY_OPERATORS_POSTFIX = ["%"];
 
 const ASSOCIATIVE_OPERATORS = ["*", "+", "&"];
 
+interface RichToken extends Token {
```

Comment:
question: is  `RichToken` richer than `EnrichedToken` ? and by how much ? :smile: :money_with_wings:

Reply:
RichToken is actually poorer than EnrichedToken ! (I can rename it TokenWithIndex or something if you want)

Reply:
RichToken > MiddleClassToken
EnrichedToken > OnePercentToken
?